### PR TITLE
Add default mode option to settings

### PR DESCRIPTION
**Pull request by basicbloke**
# Description

Add default mode to settings files

## Type of Change

- [*] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [*] I stared at the code and made sure it makes sense
- [*] Tested functionality within UI

**Test Configuration**:
Python: 3.11.9

## Checklist:

- [*] I have performed a self-review of my code
- [*] My changes generate no new warnings

### CODE REVIEW

File: private_gpt/settings/settings.py [hunk 1]
```diff
@@ -350,6 +350,10 @@ class AzureOpenAISettings(BaseModel):
 class UISettings(BaseModel):
     enabled: bool
     path: str
+    default_mode: str = Field(
+        "RAG",
+        description="The default mode.",
+    )
```
Comment (jaluma) [inline private_gpt/settings/settings.py:356]: It's better idea configure like literal (enum)

**Review by jaluma — CHANGES_REQUESTED** [review]
Please check my comments. Also, can you add this to `settings.yaml`?

**Comment by basicbloke** [comment]
need to run tests and fix

File: settings.yaml [hunk 2]
```diff
@@ -25,21 +25,23 @@ data:
 ui:
   enabled: true
   path: /
+  # "RAG", "Search", "Basic", or "Summarize"
+  default_mode: "Summarize"
```
Comment (jaluma) [inline settings.yaml:29]: By default, RAG please :)

**Review by jaluma — CHANGES_REQUESTED** [review]

File: settings.yaml (hunk 2 above)
Comment (basicbloke) [reply settings.yaml]: 😎

**Review by jaluma — APPROVED** [review]
LGTM

Thank you!

**Review by pabloogc — APPROVED** [review]

**Pull request merged**